whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/dao/Nrtm4ClientInfoRepository.java

```diff
@@ -52,15 +61,25 @@ SELECT id, source, MAX(version), session_id, type, hostname, created
             GROUP BY source
             """;
         return jdbcSlaveTemplate.query(sql,
-                (rs, rn) -> new NrtmClientVersionInfo(
-                        rs.getLong(1),
-                        rs.getString(2),
-                        rs.getLong(3),
-                        rs.getString(4),
-                        NrtmClientDocumentType.fromValue(rs.getString(5)),
-                        rs.getString(6),
-                        rs.getLong(7)
-                        ), NrtmClientDocumentType.NOTIFICATION.getFileNamePrefix());
+                nrtmClientVersionRowMapper(),
+                NrtmClientDocumentType.NOTIFICATION.getFileNamePrefix());
+    }
+
+    @Nullable
+    public NrtmClientVersionInfo getNrtmLastVersionInfoForDeltasPerSource(final String source){
+        final String sql = """
```

Comment:
Do you need to define a String here? It's only used once, you can put it in the method call directly.